docs/DeveloperGuide.md

```diff
@@ -1,9 +1,263 @@
+
+Given below is an example usage scenario and how deleting resource(s) works.
+
+**Step 1**:
+The user launches the application. The application state is initialised by the Storage class which reads the text file.
+
+**Step 2**:
+The user executes the following command to delete the 2nd resource in the resource list of the project 'CS2113'.
+>delete p/CS2113 i/2
+
+**Step 3**:
+If PROJECT_NAME is not found in the database, the application prompts the user that the project is not found. 
+<br> If INDEX is invalid <i>(i.e. negative integer or exceeds the number of resources in that project)</i>, the application prompts the user to reenter a valid index.
+<br> Else, the system feedbacks to the user that the specified resource has been successfully deleted.
+
+**Step 4**:
+The user can execute `list p/CS2113` to verify that the specified resource has been deleted from the project.
+
+The following sequence diagram shows how the delete operation works:
+![delete_puml](puml_img/deleteResource.png)
+
+### Editing resource from a specified project
+**Proposed Implementation**
+
+Given below is an example usage scenario and how editing a resource works.
+
+**Step 1**:
+The user launches the application. The application state is initialised by the Storage class which reads the text file.
+
+**Step 2**:
+The user executes the following command to edit the url and link description of the 3nd resource in the resource list of the project 'CS2113'.
+>edit p/CS2113 i/2 url/www.CS2113tp.com d/This is tp site.
+
+**Step 3**:
+If PROJECT_NAME is not found in the database, the application prompts the user that the project is not found.
+<br> If INDEX is invalid <i>(i.e. negative integer or exceeds the number of resources in that project)</i>, the application prompts the user to reenter a valid index.
+<br> Else, the system feedbacks to the user that the specified resource has been successfully edited.
+
+**Step 4**:
+The user can execute `list p/CS2113` to verify that the specified resource has been successfully edited.
+
+
+
+### Finding resource(s) in a project or all projects based on a keyword
+**Proposed Implementation**
+
+Given below is an example usage scenario and how the find mechanism behaves at each step.
+
+**Step 1**:
+The user initialises the application and adds the resources, Project and Resource objects are created accordingly.
+
+**Step 2**:
+The user wants to find resources that match a keyword website in his project CS2113. 
+He executes the following command to print out a list of resources in CS2113 that match the input keyword.
+> find k/website p/CS2113
+
+**Step 3**:
+If the project name cannot be found, the user will be prompted to search for a valid project name.
+If the project name is not given, such as in the following command, all projects will be searched to find resources matching the keyword. 
+> find k/website
+
+The following sequence diagram shows how the find operation works:
+![Exit_puml](puml_img/Find.png)
```

Comment:
Try to use a lighter color for your background so that the contrast between the words and the background is higher to improve readability. You might also want to remove the :Model class